src/main/java/com/ot/restaurant/ioc/DependencyContainer.java

```diff
@@ -0,0 +1,54 @@
+package com.ot.restaurant.ioc;
+
+import java.io.File;
+
+public class DependencyContainer<T> {
+  private final IoCContainer ioc = IoCContainer.getInstance();
+
+  private final Object entityType = new Object();
+
+  public DependencyContainer() {}
+
+  public void registerRepositories() throws Exception {
+    String entitiesFolderPath = "src/main/java/com/ot/restaurant/repositories";
+
+    File entitiesFolder = new File(entitiesFolderPath);
+
+    for (File file : entitiesFolder.listFiles()) {
+      if (file.getName().endsWith("Impl.java")) {
+        // System.out.println(file.getName());
+      } else {
+        String entityName = file.getName().substring(0, file.getName().lastIndexOf("."));
+        // System.out.println(entityName);
+
+        System.out.println("Instatiating: " + entityName);
+        ioc.register(entityName, Class.forName("com.ot.restaurant.repositories." + entityName));
+      }
+    }
+
+    // System.out.println(ioc.resolve("CustomerRepository"));
+
+  }
+
+  public void registerControllers() throws Exception {
```

Comment:
Do this manaually please, what is we create UserControllerV2 class? this could make a mistake, we should be able to decide which instance to use under the same ioc container name